core/coredatainit.go

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @author: Tingyu Zeng, Dell
+ * @version: 1.0.0
+ *******************************************************************************/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/dghubble/sling"
+)
+
```

Comment:
In this file, I would recommend genericizing the names. There's nothing in here specific to core-data except for `config.SecretService.CoredataSecretPath` -- which could conceivably be passed in from the caller. The way this is written, you will have a lot of duplicate code when this is extrapolated to all of the other services. To take another example, the `CoredataCredentials` are just credentials. They don't have anything that is essential to core-data.